src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GenericTypesTemplateMap.cs

```diff
@@ -136,8 +122,11 @@ public static bool IsArrayTypeEligibleForTemplate(TypeDesc type)
             return !elementType.IsPointer && !elementType.IsFunctionPointer;
         }
 
-        public static bool IsEligibleToHaveATemplate(TypeDesc type)
+        public static bool IsEligibleToHaveATemplate(NodeFactory factory, TypeDesc type)
         {
+            if (type.HasSameTypeDefinition(factory.ArrayOfTClass))
```

Comment:
I'm a bit suspicious about this. This will get called from `CodeBasedDependencyAlgorithm.cs` after we compile a method on `Array<T>`. Is "do nothing" the right course of action? Should that be another place where we need to compensate for the fact that virtual methods on array types don't report their owning type right?

Makes me want to bring the "reparented method" (https://github.com/MichalStrehovsky/corert/commit/5ab83c990e7678af48b7c3253a47623c548c5b29) back to the table.